src/posix/Makefile-posix

```diff
@@ -30,91 +30,63 @@
 
 .NOTPARALLEL:
 
-COVERAGE                       ?= 0
-DEBUG                          ?= 0
-DISABLE_BUILTIN_MBEDTLS        ?= 0
-DISABLE_EXECUTABLE             ?= 0
-SNTP_CLIENT                    ?= 0
+COVERAGE                             ?= 0
+DEBUG                                ?= 0
+
+# Enable most features by default to cover most code
+
+BORDER_AGENT                         ?= 1
+BORDER_ROUTER                        ?= 1
+CERT_LOG                             ?= 1
+COAP                                 ?= 1
+COAPS                                ?= 1
+COMMISSIONER                         ?= 1
+CHANNEL_MANAGER                      ?= 1
+CHANNEL_MONITOR                      ?= 1
+CHILD_SUPERVISION                    ?= 1
+DHCP6_CLIENT                         ?= 1
+DHCP6_SERVER                         ?= 1
+DIAGNOSTIC                           ?= 1
+DNS_CLIENT                           ?= 1
+ECDSA                                ?= 1
+JAM_DETECTION                        ?= 1
+JOINER                               ?= 1
+LEGACY                               ?= 1
+LINK_RAW                             ?= 0
+MAC_FILTER                           ?= 1
+MTD_NETDIAG                          ?= 1
+SERVICE                              ?= 1
+SNTP_CLIENT                          ?= 1
+UDP_FORWARD                          ?= 1
 
-ECHO                           := @echo
-MAKE                           := make
-MKDIR_P                        := mkdir -p
-LN_S                           := ln -s
-RM_F                           := rm -f
-
-BuildJobs                      ?= 10
+COMMONCFLAGS                         := \
+    -O2                                 \
```

Comment:
Is there a reason for using different optimization flags for POSIX and POSIX radio only target?